File: problema2Hireline.js
const fs = require('fs');
stdin = process.openStdin();

input = '';

stdin.on('data', (chunk) => input += chunk)
stdin.on('end', () => calculate());

const calculate = () => {
    const lines = input.split('\n');
    const n = parseInt(lines.shift);

    let distance = 0;
    lines.forEach(line => {
        const turn = line.split(' ');
        const diff = turn[0] - turn[1];
        if (Math.abs(diff) > Math.abs(distance)) distance = diff;
    })

    let response;
    if (distance > 0) response = '1 ' + distance;
    else response = '2 ' + -distance;

    fs.writeFile('output.txt', response, (err) => err ? console.log(err) : []);
}

File: problema2Hireline.py
import sys

lines = []

for line in sys.stdin:
    lines.append("".join(line.splitlines()))

n = int(lines.pop(0))
distance = 0

for i in range(n):
    turn = [int(number) for number in lines.pop(0).split()]
    diff = turn[0] - turn[1]
    if abs(diff) > abs(distance):
        distance = diff

if distance > 0:
    response = '1 ' + str(distance)
else:
    response = '2 ' + str(-distance)

file = open('output.txt', 'w')
file.write(response)